Movie-App-Group-7/Movie-App-Group-7/UI/Detail Movie/ViewModel/DetailMovieViewModel.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  DetailMovieViewModel.swift
+//  Movie-App-Group-7
+//
+//  Created by Mohammad Azri on 20/10/22.
+//
+
+import Foundation
+
+enum RetrievingDetailMovieState {
+    case success(Movie?)
+    case failure(String)
+}
+
+class DetailMovieViewModel {
+    private let movieNetworkModel : MovieNetworkModel
+    
+    init(movieNetworkModel: MovieNetworkModel) {
+        self.movieNetworkModel = movieNetworkModel
+    }
+    
+    func retrieveMovie(movieId: String, completion : @escaping (RetrievingDetailMovieState) -> ()) {
+        movieNetworkModel.getMovie(movieId: movieId) { result in
+            switch result {
+            case .success(let movie):
+                completion(.success(movie))
+            case .failure(let message):
+                completion(.failure(message))
+            }
+        }
+        
+    }
+}
```

Comment:
Klo niat lu implement MVVM, lu gabisa bilang ini ViewModel, karena viewModel di MVVM itu punya binder/observer yang menjadi pembeda itu sama MVC. lu harus bikin binder/observernya terus di observer di view, instead of lu handle from escaping when calling retrieveMovie(..). jadi Klo MVVM gamake result dari Completion ini jadi ga perlu ada completion di function retrieveMovie(..).

**Ini berlaku untuk semua ViewModel yang lu implementasikan Project ini**  

```suggestion
class DetailMovieViewModel {
    private let movieNetworkModel : MovieNetworkModel
    private var resultsObserver : ((Movie) -> ()?) // nanti ini yg lu observer di Controllernya
    private var errorObserver : ((String) -> ()?)  // nanti ini yg lu observer di Controllernya
    
    init(movieNetworkModel: MovieNetworkModel) {
        self.movieNetworkModel = movieNetworkModel
    }
    
    func retrieveMovie(movieId: String) {
        movieNetworkModel.getMovie(movieId: movieId) { result in
            switch result {
            case .success(let movie):
            resultsObserver?(movie)
            case .failure(let message):
            errorObserver?(message)
            }
        }
        
    }
}
```